source/time_stepping/RSF_convection_time_step.cc

```diff
@@ -0,0 +1,143 @@
+                  }
+              }
+
+            max_local_speed_over_meshsize = std::max(max_local_speed_over_meshsize,
+                                                     max_local_velocity
+                                                     /
+                                                     cell->minimum_vertex_distance());
+            max_local_speed_over_meshsize_old = std::max(max_local_speed_over_meshsize_old,
+                                                         max_local_velocity_old
+                                                         /
+                                                         cell->minimum_vertex_distance());
+
+          }
+
+      const double max_global_speed_over_meshsize
+        = Utilities::MPI::max (max_local_speed_over_meshsize, this->get_mpi_communicator());
+      const double max_global_speed_over_meshsize_old
+        = Utilities::MPI::max (max_local_speed_over_meshsize_old, this->get_mpi_communicator());
+
+      double min_RSF_convection_timestep = std::numeric_limits<double>::max();
+
+      // if max velocities stayed the same or decreased, take the CFL number specified in the input file.
+      // else, take 0.5 as the CFL number to let the timestep increase faster
+      if ((max_global_speed_over_meshsize != 0.0) && (max_global_speed_over_meshsize <= max_global_speed_over_meshsize_old))
+        min_RSF_convection_timestep = this->get_parameters().CFL_number / (this->get_parameters().temperature_degree * max_global_speed_over_meshsize);
+      else
+        min_RSF_convection_timestep = 0.5 / (this->get_parameters().temperature_degree * max_global_speed_over_meshsize);
+
+      AssertThrow (min_RSF_convection_timestep > 0,
+                   ExcMessage("The time step length for the each time step needs to be positive, "
+                              "but the computed step length was: " + std::to_string(min_RSF_convection_timestep) + ". "
+                              "Please check for non-positive material properties."));
+
+      return min_RSF_convection_timestep;
+    }
+
+
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    ASPECT_REGISTER_TIME_STEPPING_MODEL(RSFConvectionTimeStep,
+                                        "RSF convection time step",
+                                        "This model computes the convection time step as "
+                                        "$ CFL / \\max \\| u \\| / h$ over all cells, "
+                                        "where $u$ is the velocity and $h$ is the product of mesh size "
+                                        "and temperature polynomial degree. Depending on the current "
+                                        "and previous velocity, different CFL numbers are used to "
+                                        "increase timestep size faster after an earthquake. This is "
+                                        "only recommended for rate-and-state friction models.")
```

Comment:
In this case, you should throw when RSF is not used.

Reply:
good point, thanks!